BUILD.bazel

```diff
@@ -71,6 +71,7 @@ gazelle_binary(
 
 # gazelle:prefix github.com/cgrindel/rules_swift_package_manager
 # gazelle:go_naming_convention import
+# gazelle:go_naming_convention_external go_default_library
```

Comment:
Can you say why this line is necessary? It looks like a bug in Gazelle if it is.

Reply:
If I remove the `gazelle:go_naming_convention_external`, some (not all) of the external dep are stripped of the `:go_default_library`, which results in Bazel analysis errors. TBH, I have not spent much time figuring out what is happening.

One point I think is peculiar is why `:go_default_library` was selected by Gazelle for the external deps when I configured `import` for the local Go targets. I would have thought that it would have inferred that we wanted to use `import` for all of the Go targerts.

Reply:
Yes, that's also what worries me here. With Bzlmod, Gazelle should reliably take the build naming conventions of external repos into account. 

I will take a closer look.

Reply:
I can't reproduce a problem anymore - can you provide me with steps that result in Gazelle generating broken references to external repos?

Reply:
Created https://github.com/bazelbuild/bazel-gazelle/issues/1557 to track the issue. It includes the repro steps.